src/backend/api/Fusion.Resources.Api/Authorization/Handlers/ProjectAccessAuthHandler.cs

```diff
@@ -33,9 +33,9 @@ private async Task VerifyManageContractsAsync(AuthorizationHandlerContext contex
                 // User must work in procurement position in the project
                 var profile = await profileResolver.GetCurrentUserFullProfileAsync();
 
-                var positions = profile.Positions.Where(p => p.Project.Id == resource.ProjectId);
+                var positions = profile.Positions?.Where(p => p.Project.Id == resource.ProjectId);
                 
-                if (positions.Any(p => p.BasePosition.Discipline == "Procurement"))
+                if (positions != null && positions.Any(p => p.BasePosition.Discipline == "Procurement"))
```

Comment:
check added because a profile may not contain any position(s)
(At least according to line above, which is: 
var positions = profile.Positions?.Where(p => p.Project.Id == resource.ProjectId);